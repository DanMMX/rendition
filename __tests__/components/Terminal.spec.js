/* globals expect, jest, test */
import React from 'react'
import renderer from 'react-test-renderer'
import Provider from '../../src/components/Provider'
import Terminal from '../../src/components/Terminal'

// Terminal implements xterm.js v3, which uses window.matchMedia
// This API is not supported by jsDom, so we have to mock it here
const matchMediaMock = require('match-media-mock').create()
matchMediaMock.setConfig({type: 'screen', width: 1200})

Object.defineProperty(window, 'matchMedia', {
  value: jest.fn(matchMediaMock)
})

test('Terminal renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Terminal>Terminal</Terminal>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
