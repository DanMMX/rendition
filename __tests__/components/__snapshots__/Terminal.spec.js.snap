// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terminal renders correctly 1`] = `
<div
  className="sc-bdVaJa jKopLr"
>
  <div
    className="sc-VigVT dPbUyP sc-gqjmRU bCjhLA sc-iwsKbI khHCQa sc-bwzfXH ihgQcP"
  >
    <div
      className="sc-jTzLTM kUYmvY"
    />
  </div>
</div>
`;
